cloud/aws/deploy/website.go

```diff
@@ -217,33 +219,123 @@ func (a *NitricAwsPulumiProvider) deployCloudfrontDistribution(ctx *pulumi.Conte
 		return err
 	}
 
-	policy := pulumi.All(a.publicWebsiteBucket.Arn, oai.IamArn).ApplyT(func(args []interface{}) (string, error) {
-		bucketID, bucketIdOk := args[0].(string)
-		oaiPath, oaiPathOk := args[1].(string)
-		if !bucketIdOk || !oaiPathOk {
-			return "", fmt.Errorf("failed to get bucket ID or OAI path")
+	for websiteName, website := range a.Websites {
+		policy := pulumi.All(website.bucket.Arn, oai.IamArn).ApplyT(func(args []interface{}) (string, error) {
+			bucketID, bucketIdOk := args[0].(string)
+			oaiPath, oaiPathOk := args[1].(string)
+
+			if !bucketIdOk || !oaiPathOk {
+				return "", fmt.Errorf("failed to get bucket ID or OAI path")
+			}
+
+			return fmt.Sprintf(`{
+				"Version": "2012-10-17",
+				"Statement": [
+					{
+						"Effect": "Allow",
+						"Principal": {
+							"AWS": "%s"
+						},
+						"Action": "s3:GetObject",
+						"Resource": "%s/*"
+					}
+				]
+			}`, oaiPath, bucketID), nil
+		}).(pulumi.StringOutput)
+
+		_, err = s3.NewBucketPolicy(ctx, fmt.Sprintf("bucket-policy-%s", websiteName), &s3.BucketPolicyArgs{
+			Bucket: website.bucket.Bucket,
+			Policy: policy,
+		})
+		if err != nil {
+			return err
 		}
-		return fmt.Sprintf(`{
-			"Version": "2012-10-17",
-			"Statement": [
-				{
-					"Effect": "Allow",
-					"Principal": {
-						"AWS": "%s"
-					},
-					"Action": "s3:GetObject",
-					"Resource": "%s/*"
-				}
-			]
-		}`, oaiPath, bucketID), nil
-	}).(pulumi.StringOutput)
 
-	_, err = s3.NewBucketPolicy(ctx, "publicBucketPolicy", &s3.BucketPolicyArgs{
-		Bucket: a.publicWebsiteBucket.Bucket,
-		Policy: policy,
-	})
-	if err != nil {
-		return err
+		code, err := embeds.GetUrlRewriteFunction(website.basePath)
+		if err != nil {
+			return err
+		}
+
+		rewriteFun, err := cloudfront.NewFunction(ctx, fmt.Sprintf("url-rewrite-function-%s", websiteName), &cloudfront.FunctionArgs{
+			Comment: pulumi.String("Rewrite URLs to default index document"),
+			Code:    code,
+			Runtime: pulumi.String("cloudfront-js-1.0"),
+		})
+		if err != nil {
+			return err
+		}
+
+		origins = append(origins, &cloudfront.DistributionOriginArgs{
+			DomainName: website.bucket.BucketRegionalDomainName,
+			OriginId:   pulumi.String(websiteName),
+			S3OriginConfig: cloudfront.DistributionOriginS3OriginConfigArgs{
+				OriginAccessIdentity: oai.CloudfrontAccessIdentityPath,
+			},
+		})
+
+		// Make cache behaviour for all but the root origin. The root origin uses the default cache behaviour
+		if website.basePath != "/" {
+			orderedCacheBehaviors = append(orderedCacheBehaviors,
+				&cloudfront.DistributionOrderedCacheBehaviorArgs{
+					PathPattern: pulumi.Sprintf("%s*", strings.TrimPrefix(website.basePath, "/")),
```

Comment:
for subsites this would potentially trigger if a main site page starts with the subsite path?